control_server.go

```diff
@@ -136,3 +175,80 @@ func (this *ControlServer) HandleVerify(w http.ResponseWriter, r *http.Request)
 
 	http.Redirect(w, r, "/", http.StatusSeeOther)
 }
+
+func (this *ControlServer) fetchStatus() *ControlServerStatus {
+	status := &ControlServerStatus{}
+	status.Progress = *(this.F.Progress())
+
+	status.GhostferryVersion = VersionString
+
+	status.SourceHostPort = fmt.Sprintf("%s:%d", this.F.Source.Host, this.F.Source.Port)
+	status.TargetHostPort = fmt.Sprintf("%s:%d", this.F.Target.Host, this.F.Target.Port)
+
+	status.OverallState = fmt.Sprintf("%s", this.F.OverallState.Load())
+
+	status.StartTime = this.F.StartTime
+	status.CurrentTime = time.Now()
+	status.TimeTaken = time.Duration(status.TimeTaken)
+
+	status.BinlogStreamerLag = time.Duration(status.BinlogStreamerLag)
+
+	status.AutomaticCutover = this.F.Config.AutomaticCutover
+	status.BinlogStreamerStopRequested = this.F.BinlogStreamer.stopRequested
+
+	// Getting all table statuses
+	status.TableStatuses = make([]*TableStatus, 0, len(status.Tables))
+	status.CompletedTableCount = 0
+	status.TotalTableCount = len(status.Tables)
+	status.AllTableNames = make([]string, 0, len(status.Tables))
+
+	dbSet := make(map[string]bool)
+	for name, tableProgress := range status.Tables {
```

Comment:
Question: can we use `status.Tables` directly?

Reply:
Umm status.Tables contains mapping of table names to `TableProgress` which is sort of similar to `ControlServerTableStatus`. but there are some fields missing there which we need in `ControlServerTableStatus` so I just decided to keep this instead of using `status.Tables` directly.